storage/storage_dest.go

```diff
@@ -335,6 +335,11 @@ func (s *storageImageDestination) PutBlobPartial(ctx context.Context, chunkAcces
 
 	out, err := s.imageRef.transport.store.PrepareStagedLayer(nil, differ)
 	if err != nil {
+		// Special-case the error message
+		var perr chunked.ErrFallbackToOrdinaryLayerDownload
+		if errors.Is(err, storage.ErrNotSupported) && !errors.As(err, &perr) {
+			err = fmt.Errorf("partial layer pull is not supported by the graph driver, and fallback is not allowed: %w", err)
```

Comment:
Could we list the graph driver in the error?

Reply:
Meanwhile I have prepared https://github.com/containers/storage/pull/2140; that is where the error is going to come from in the future.

And c/image does not authoritatively know that the graph driver is the cause.

So I think I’ll simplify this again to just add more context without special-casing.